src/operators/aggregate.cc

```diff
@@ -570,6 +570,23 @@ void Aggregate::SortResult(std::vector<arrow::Datum>& groups,
   }
 }
 
+void Aggregate::Clear() {
+  num_aggs_ = 0;
+
+  prev_result_ = nullptr;
```

Comment:
please use the .reset() method it is more idiomatic

Also (without knowing if it makes an actual difference = nullptr and .reset in terms of operations needed. There is some casting and conversion happening when - nullptr is used
https://stackoverflow.com/questions/16151550/c11-when-clearing-shared-ptr-should-i-use-reset-or-set-to-nullptr/46577399#46577399